app/code/core/Mage/Eav/Model/Entity/Type.php

```diff
@@ -99,24 +99,33 @@ public function loadByCode($code)
     /**
      * Retrieve entity type attributes collection
      *
-     * @param   int $setId
-     * @return  Mage_Eav_Model_Resource_Entity_Attribute_Collection
+     * @param int|null $setId
+     * @param bool $useCache reuse local cache for collection
+     * @return Mage_Eav_Model_Resource_Entity_Attribute_Collection
      */
-    public function getAttributeCollection($setId = null)
+    public function getAttributeCollection($setId = null, $useCache = true)
```

Comment:
do we have to change the signature of this public method?

Reply:
If I want to re-use the code, mostly yes. But since it's an optional parameter it should not create any problems.
Alternatively I can refactor the code into a second method `newAttributeCollection` that always returns a fresh one, which is then used by `getAttributeCollection` as needed. Which do you prefer?

Reply:
the thing is (but I wanted to have other people join the conversation) that modifying the signature of a public method should be considered a major change, since all classes extending that method (which, ok, most probably are zero in the world) will break

Reply:
I think changing the signature is okay in this case, especially since we're still on RC for v20.1.0. The error should appear very quickly upon testing if anyone has override the method, and it's a quick fix.

Generally though, if it makes more sense to introduce a new method, that would be preferable in most cases. But, we also don't want to litter the code with slight variations of similar method names to avoid this problem.

Reply:
do we need the useCache param tho? it seems that's just used in the initialization, isn't it possible to find a way to detect that we are in the initialization phase without the param? maybe I'm just blabbering

Reply:
nothing magic, if some property doesn't exists it means cache was not loaded etc..

Reply:
something like

<img width="911" alt="Screenshot 2023-03-09 alle 14 38 09" src="https://user-images.githubusercontent.com/909743/224057949-994457ad-8873-45bc-870f-107a320848b8.png">

Reply:
I know, but it's a delicate problem - you cannot predict what someone else does to a "cached" and therefor shared collection downstream, which caused the problem with for example adding store labels to it that triggered the backend exception when multiple scopes were initialized in a single context.